msgraph-mail/go/utilities/models/attachment.go

```diff
@@ -38,54 +39,19 @@ func (m *Attachment) GetContentType()(*string) {
 func (m *Attachment) GetFieldDeserializers()(map[string]func(i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode)(error)) {
     res := m.Entity.GetFieldDeserializers()
     res["contentType"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
-        val, err := n.GetStringValue()
-        if err != nil {
-            return err
-        }
-        if val != nil {
-            m.SetContentType(val)
-        }
-        return nil
+        return core.SetValue(n.GetStringValue , m.SetContentType)
     }
     res["isInline"] = func (n i878a80d2330e89d26896388a3f487eef27b0a0e6c010c493bf80be1452208f91.ParseNode) error {
-        val, err := n.GetBoolValue()
-        if err != nil {
-            return err
-        }
-        if val != nil {
-            m.SetIsInline(val)
-        }
-        return nil
+        return core.SetValue(n.GetBoolValue , m.SetIsInline)
```

Comment:
couldn't we have the helper method return a function that matches the prototype? This way we could have something like:

```go
res["isInline"] = core.SetBooleanValue(m.SetIsInline)
```

and the SetBooleanValue method would have the following prototype

```go
func SetValue[T](setter func (T) void)) func (ParseNode) (error) {}
```

Reply:
Thought about it, but we would not be able to know which method to call from `ParseNode`

Reply:
Great idea! This will definitely work

Reply:
done